src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -12,29 +13,30 @@ internal static partial class ThrowHelper
         // If the exception source is this value, the serializer will re-throw as JsonException.
         public const string ExceptionSourceValueToRethrowAsJsonException = "System.Text.Json.Rethrowable";
 
-        public static ArgumentOutOfRangeException GetArgumentOutOfRangeException_MaxDepthMustBePositive(string parameterName)
+        public static ArgumentOutOfRangeException GetArgumentOutOfRangeException_MaxDepthMustBePositive(string? parameterName)
```

Comment:
Why are these parameters nullable on the ThrowHelper APIs?

Reply:
I would pharase it is allowing null not need to null. In general everything we do annotate nullable if it allowing null

Reply:
Updated all types called with non null value to be non null